src/bullets/data_source/data_source_fmp.py

```diff
@@ -12,40 +14,96 @@ def __init__(self, token: str, resolution: Resolution):
         super().__init__()
         self.token = token
         self.resolution = resolution
+        self.stocks = {}
+
+    def store_price_points(self, symbol: str, start_date: datetime.date, end_date: datetime.date = None,
+                           limit: int = 1000):
+        if symbol in self.stocks:
+            stock = self.stocks[symbol]
+        else:
+            stock = Stock(symbol, self.resolution)
 
+        if self.resolution == Resolution.DAILY:
+            url_resolution = "historical-price-full/"
+        else:
+            url_resolution = "historical-chart/" + str(self.resolution.value[0]) + "/"
 
-    def get_price(self, symbol: str, timestamp:datetime = None):
+        if end_date is None:
+            nb_entree = limit
+            if self.resolution == Resolution.HOURLY:
+                nb_entree = math.ceil(nb_entree/6)
+            elif self.resolution == Resolution.MINUTE:
+                nb_entree = math.ceil(nb_entree/390)
+
+            end_date = start_date + datetime.timedelta(days=nb_entree)
+            if end_date > datetime.date.today():
+                end_date = datetime.date.today()
+
+        current_end_date = end_date
+        finished = False
+
+        while not finished:
+            interval = "from=" + str(start_date) + "&to=" + str(current_end_date)
+            url = self.URL_BASE_FMP + url_resolution + symbol + "?" + interval + "&apikey=" + self.token
+            response = self.request(url)
+
+            if response == "{ }":
+                break
+            result = json.loads(response)
+            if self.resolution == Resolution.DAILY:
+                data = result["historical"]
+            else:
+                data = result
+
+            first_date = current_end_date
+
+            for entry in data:
+                price_point = PricePoint(entry)
+                if self.resolution == Resolution.DAILY:
+                    stock_date = datetime.datetime.strptime(price_point.date + " 00:00:00", "%Y-%m-%d %H:%M:%S")
+                else:
+                    stock_date = datetime.datetime.strptime(price_point.date, "%Y-%m-%d %H:%M:%S")
+
+                stock.price_points[stock_date] = price_point
+                first_date = stock_date.date()
+
+            current_end_date = first_date
+            if first_date <= start_date:
+                finished = True
+
+        self.stocks[symbol] = stock
+
+    def get_market_open_close(self):
```

Comment:
The logic here is good, I'm just wondering if we should rely on the FMP endpoint. Since it doesn't provide enough data in the past, we'll still need to find another way to fix the holiday issue.

Maybe we should take this logic out of this pull request and re-evaluate what we need to do in the holiday issue.